src/carousel/carousel.js

```diff
@@ -421,11 +420,24 @@ angular.module('ui.bootstrap.carousel')
 
 .value('$carouselSuppressWarning', false)
 
+.controller('CarouselController', ['$scope', '$element', '$controller', '$log', '$carouselSuppressWarning', function($scope, $element, $controller, $log, $carouselSuppressWarning) {
+  if (!$carouselSuppressWarning) {
+    $log.warn('CarouselController is now deprecated. Use UibCarouselController instead.');
+  }
+
+  angular.extend(this, $controller('UibCarouselController', {
+    $scope: $scope,
+    $element: $element
+  }));
+}])
+
 .directive('carousel', ['$log', '$carouselSuppressWarning', function($log, $carouselSuppressWarning) {
   return {
-    restrict: 'EA',
     transclude: true,
     replace: true,
+    controller: 'CarouselController',
+    controllerAs: 'carousel',
```

Comment:
We started to add this option to some directives to expose the controller to the template using `controllerAs`. It is not something I would add personally, but there it is.